ksa_compliance/jinja.py

```diff
@@ -1,2 +1,81 @@
-def get_zatca_qr_for_invoice(invoice_name: str) -> str:
-    pass
+import base64
+
+import frappe
+from io import BytesIO
+from frappe.utils.data import add_to_date, get_time, getdate
+import datetime
+from base64 import b64encode, b64decode
+import pyqrcode
+
+
+@frappe.whitelist()
+def get_zatca_phase_1_qr_for_invoice(invoice_name: str) -> str:
+    values_dict = get_qr_inputs(invoice_name)
+    decoded_string = generate_decoded_string(values_dict)
+    print("decoded_string",decoded_string)
+    return generate_qrcode(decoded_string)
+
+
+def get_qr_inputs(invoice_name):
+    sales_invoice = frappe.get_doc("Sales Invoice", invoice_name)
```

Comment:
Please use `cast` to enable type analysis, e.g. `cast(SalesInvoice, frappe.get_doc("Sales Invoice", invoice_name))`